### DISCUSSION TITLE
Voice control of all devices in domain without specifying area?

### DISCUSSION THREAD

**Discussion by seancmalloy** [OHF-Voice/intents#1669] [Supported Sentences]
Is it possible to control all devices in domain (specifically light) without specifying area? I can control lights based on their location but see that there is no way to do this yet. I started creating custom  intents/sentences for this but lack knowledge of how this could work. 

So far: created custom_sentences/en directory with a few yamls using HassTurnOn & HassTurnOff

language: "en"
intents:
  HassTurnOn:
    data:
      - sentences:
          - "<turn> on [all] <light>"
        requires_context:
          domain: "light"
        slots:
          domain: "light"
          name: "all"
          response: "light"

update - so i guess ill just use custom group of lights named "all lights" with new area named "here" = "turn on all lights in here"

update - better yet, group named "lights" area named "All" = "turn on (or off) all lights"

but i still would like to call devices without area, i would appreciate any help

**Comment by tetele** [comment]
Do you mean like this, based on the voice satellite assigned area? https://github.com/home-assistant/core/pull/103939

If so, that's coming in 2023.12 without you having to do anything

**Comment by seancmalloy** [comment]
I don't believe so, this looks like it still uses areas, but definitely looks useful. I was looking for a way to voice control all lights which are in different areas.  So I could just say "turn off all lights" instead of  "turn off all lights in bedroom" + "turn off all lights in living room" + "turn off all lights bathroom" + ... + ... + ... 

I did find a workaround.

**Reply by tetele** [reply]
Maybe "[turn off the lights everywhere](https://github.com/home-assistant/intents/blob/c66ed53f7422f2d082fd3df1057ace09b9e3367a/sentences/en/light_HassTurnOff.yaml#L25)" is what you're looking for, then?

Admittedly, "turn off all lights [everywhere]" should be there as well.

**Comment by seancmalloy** [comment]
Yes, exactly.  I was refencing light_HassTurnOn.yaml so did not see that. With this I WAS able to create custom sentences to do it. This does indeed controls all lights regardless of area, although may need to modified if satellites are added. Thanks.

Tested to work-

![Screenshot (51)](https://github.com/home-assistant/intents/assets/23019226/e87a83f6-9d7b-4203-a9ea-4b5775ff6a02)

![Screenshot (52)](https://github.com/home-assistant/intents/assets/23019226/b4019464-f07a-4221-8be7-72e2fb00a13a)


But, can this be done for brightness and color? Yes

![Screenshot (53)](https://github.com/home-assistant/intents/assets/23019226/3e8c3bec-5b3d-4d7e-a0f5-ece94b713f21)
